### ISSUE TITLE
importing JSON files

### ISSUE THREAD

**Issue by mvolkmann** [jakedeichert/svelvet#58]
I’m familiar with modifying `rollup.config.js` and registering a plugin to support importing `.json` files. If I use svelvet, will my `.svelte` files be able to import `.json` files like this?

```js
import someData from ‘./some.json’;
```

**Comment by jakedeichert** [comment]
This I think is actually possible if you add a `babel.config.js` to your project and use a plugin like  [babel-plugin-inline-json-import](https://www.npmjs.com/package/babel-plugin-inline-json-import).

Make sure you copy this [babel.config.js](https://github.com/jakedeichert/svelvet/blob/master/tests/snapshot-babel-override/babel.config.js) as your starting point.

Note that svelvet was designed to not run your code through any bundler process... so we don't have css/json/image loaders built in. But since we do run all code through babel, this case happens to be possible 👍

**Comment by mvolkmann** [comment]
@jakedeichert I'm not getting this to work. I have a basic example at https://github.com/mvolkmann/svelvet-demo/tree/master. I think my `babel.config.js` is correct. The file `src/App.svelte` imports a JSON file. When I run `npm run dev` I get this error:

Failed to transform with babel: dist/App.js
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined

Do you see anything wrong with my `babel.config.js` or the way I'm importing the JSON file in `src/App.svelte`?

**Comment by jakedeichert** [comment]
@mvolkmann i just gave this a try and i think there's a bug in that plugin. On [this line](https://github.com/yggie/babel-plugin-inline-json-import/blob/master/src/index.js#L106), `fileLocation` is `undefined` for me.

This plugin looks pretty simple.. i may find time to debug it tomorrow unless you beat me to it :)

You can probably copy/paste their [source file](https://github.com/yggie/babel-plugin-inline-json-import/blob/master/src/index.js) and then add that to your babel config to debug it locally. Pretty sure that is possible at least

**Comment by mvolkmann** [comment]
I can fix the plugin by adding one line, but I don't understand it. Here is my updated `requireModule` function from that plugin. Note the line with "RMV" in the comment:

```js
function requireModule(moduleName, state) {
  var fileLocation = state.file.opts.filename;
  var filepath = null;

  if (!fileLocation) fileLocation = 'src/whatever'; // RMV added

  if (fileLocation === 'unknown') {
    filepath = moduleName;
  } else {
    filepath = _path2.default.join(_path2.default.resolve(fileLocation), '..', moduleName);
  }

  if (filepath.slice(-5) === '!json') {
    filepath = filepath.slice(0, filepath.length - 5);
  }

  return requireFresh(filepath);
}
```

**Comment by jakedeichert** [comment]
Yeah, so you added this line `if (!fileLocation) fileLocation = 'src/whatever';` and it works? I'm guessing `whatever` is a real file inside the src directory? 

Yeah this plugin maybe just isn't working properly with the current version of node or something.

I haven't searched hard for an alternative plugin yet, but i might consider rewriting/fixing this properly.

**Comment by mvolkmann** [comment]
Surprisingly no, “whatever” is not a real directory. That gets ignored for some reason. But if I leave that off then it doesn’t work. Very odd.

**Comment by jakedeichert** [comment]
So i think your fix kind of works because the code:

```js
filepath = Path.join(Path.resolve(fileLocation), '..', moduleName)
```

Takes your `fileLocation` (`src/whatever`) and goes up one directory with `..` which results in just `src/` remaining. Then it finds the json file relative to that it seems... but it may break if you do a relative import lower down in your directory tree.

**Comment by jakedeichert** [comment]
I have a fix for this coming.. need to adjust svelvet.

**Comment by jakedeichert** [comment]
Think I fixed this! That plugin was depending on babel knowing the `filename`.. but svelvet never provided that to babel in the first place.

Just released [v0.3.6](https://github.com/jakedeichert/svelvet/releases/tag/v0.3.6) which now tells babel the path to the file being processed. Tested locally and that babel json import plugin should now work 👍

**Comment by mvolkmann** [comment]
@jakedeichert Excellent! Thanks for fixing this!
